app/assets/javascripts/components/general_button.es6.jsx

```diff
@@ -0,0 +1,58 @@
+class GeneralButton extends React.Component {
+
+  static get propTypes() {
+    return {
+      action: React.PropTypes.func,
+      route: React.PropTypes.string,
+      content: React.PropTypes.string.isRequired,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      action: null,
+      route: '',
+      content: '',
+    };
+  }
+
+  get styles() {
+    return {
+      container: {
+        display: 'inline-block',
+        verticalAlgin: 'middle',
+        position: 'relative',
+        padding: '6px 12px',
+        color: 'white',
+        backgroundColor: '#68B1DE',
+        borderBottom: '2px solid',
+        borderColor: '#28719E',
+        borderRadius: 2,
+        fontSize: '14px',
+      },
+    };
+  }
+
+  handleClick(event) {
+    if (this.props.route === '') {
```

Comment:
Essentially we can divide actions into two categories: "client" actions and "server" actions. This action just wants to bring us to a new page on button click, so it's a "client" action and doesn't need to communicate with the server. In this future this will actually change because logging in needs to talk to the server for authentication, but for now we'll treat it as a "client" action because that's not set up yet.

Other examples of "client" actions we'll have include: hiding the sidebar on button click, switching to a page that doesn't need authentication, changing the view look of cards on the page. etc/

Examples of "server" action we'll have include: authentication on sign up or log in, loading data to render on our dashboard, saving changes made with the form manager/creator, etc.